darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -0,0 +1,235 @@
+
+    def get_metadata_table(self):
+        """
+        Builds the metadata table (in html form) for the summary report.
+        """
+        # assign the metadata dictionary
+        job_data = self.report.metadata["job"]
+        # TODO: once this is exposed, add the correct entry
+        log_fmt_ver = "N/A"
+        # build a dictionary with the appropriate metadata
+        metadata_dict = {
+            "Job ID": job_data["jobid"],
+            "User ID": job_data["uid"],
+            "# Processes": job_data["nprocs"],
+            "Runtime (s)": self.get_runtime(report=self.report),
+            "Start Time": datetime.datetime.fromtimestamp(job_data["start_time"]),
+            "End Time": datetime.datetime.fromtimestamp(job_data["end_time"]),
+            "Command": self.get_full_command(report=self.report),
+            "Log Filename": os.path.basename(self.log_path),
+            "Runtime Library Version": job_data["metadata"]["lib_ver"],
+            "Log Format Version": log_fmt_ver,
+        }
+        # convert the dictionary into a dataframe
+        metadata_df = pd.DataFrame.from_dict(data=metadata_dict, orient="index")
+        # write out the table in html
+        self.metadata_table = metadata_df.to_html(header=False, border=0)
+
+    def get_module_table(self):
+        """
+        Builds the module table (in html form) for the summary report.
+        """
+        # construct a dictionary containing the module names
+        # and their respective data stored in KiB
+        module_dict = {}
+        for mod in self.report.modules:
+            # retrieve the module version and buffer sizes
+            mod_version = self.report.modules[mod]["ver"]
+            # retrieve the buffer size converted to KiB
+            mod_buf_size = self.report.modules[mod]["len"] / 1024
+            # create the key/value pairs for the dictionary
+            key = f"{mod} (ver={mod_version})"
+            val = f"{mod_buf_size:.2f} KiB"
+            module_dict[key] = val
+
+        # convert the module dictionary into a dataframe
+        module_df = pd.DataFrame.from_dict(data=module_dict, orient="index")
+        # write out the table in html
+        self.module_table = module_df.to_html(header=False, border=0)
+
+    def get_stylesheet(self):
+        """
+        Retrieves the locally stored CSS.
+        """
+        # get the path to the style sheet
+        with importlib_resources.path(darshan.cli, "style.css") as path:
+            # collect the css entries
+            with open(path, "r") as f:
+                self.stylesheet = "".join(f.readlines())
+
+
+def setup_parser(parser: Any):
+    """
+    Configures the command line arguments.
+
+    Parameters
+    ----------
+    parser : command line argument parser.
+
+    """
+    parser.add_argument(
+        "log_path",
+        type=str,
+        help="Specify path to darshan log.",
+    )
+    parser.add_argument("--output", type=str, help="Specify output filename.")
```

Comment:
I thought I asked for an issue to be opened about argparse problems. For example, the help not listing the options like `--output` here with `python -m darshan summary --help`:

```
usage: darshan <command>

PyDarshan CLI Utilities

optional arguments:
  -h, --help  show this help message and exit
  --debug
  --version
```

I'll open it now...